__tests__/app.test.js

```diff
@@ -206,6 +211,138 @@ describe('GET /api/articles/:article_id/comments', () => {
   });
 });
 
+describe('POST /api/articles/:article_id/comments', () => {
+  test('A succesful comment should respond with an appropriate status and the posted comment with correct comment properties', () => {
+    const newComment = {
+      username : 'lurker',
+      body : "body..."
+    };
+    return request(app)
+      .post('/api/articles/1/comments')
+      .send(newComment)
+      .expect(201)
+      .then((response) => {
+        const comment = response.body.comment;
+        
+        expect(comment).toHaveProperty("comment_id")
+        expect(typeof comment.comment_id).toBe("number")
+
+        expect(comment).toHaveProperty("body")
+        expect(typeof comment.body).toBe("string")
+
+        expect(comment).toHaveProperty("article_id")
+        expect(typeof comment.article_id).toBe("number")
+
+        expect(comment).toHaveProperty("author")
+        expect(typeof comment.author).toBe("string")
+
+        expect(comment).toHaveProperty("votes")
+        expect(typeof comment.votes).toBe("number")
+
+        expect(comment).toHaveProperty("created_at")
+        expect(typeof comment.created_at).toBe("string")
+      });
+  });
+  test('Sends an appropriate status and error message when given an invalid article id', () => {
+    const badComment = {
+      username : "lurker",
+      body : "body..."
+    };
+    return request(app)
+      .post('/api/articles/999/comments')
+      .send(badComment)
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe('Article does not exist');
+      });
+  });
+  test("Responds with an appropriate status and error message when provided with a username that doesn't match an existing username from the users table", () => {
+    const badComment = {
+      username : "Test",
+      body : "body..."
+    };
+    return request(app)
+      .post('/api/articles/1/comments')
+      .send(badComment)
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe('User does not exist');
+      });
+  });
+  test('Responds with an appropriate status and error message when provided with an Invalid request body', () => {
+    const badComment = {
+      user : "Test"
+    };
+    return request(app)
+      .post('/api/articles/1/comments')
+      .send(badComment)
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe('Invalid request body');
+      });
+  });
+});
```

Comment:
Really good testing, another you should add on is a test for when the _article_id_ is not a valid data type, for example `/api/articles/banana/comments`